test/Microsoft.Extensions.ML.Tests/FileLoaderTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Xunit;
+
+namespace Microsoft.Extensions.ML
+{
+    public class FileLoaderTests
+    {
+        [Fact]
+        public void throw_until_started()
+        {
+            var services = new ServiceCollection()
+                .AddOptions()
+                .AddLogging();
+            var sp = services.BuildServiceProvider();
+
+            var loaderUnderTest = ActivatorUtilities.CreateInstance<FileModelLoader>(sp);
+            Assert.Throws<InvalidOperationException>(() => loaderUnderTest.GetModel());
+            Assert.Throws<InvalidOperationException>(() => loaderUnderTest.GetReloadToken());
+        }
+
+        [Fact]
+        public void can_load_model()
+        {
+            var services = new ServiceCollection()
+                .AddOptions()
+                .AddLogging();
+            var sp = services.BuildServiceProvider();
+
+            var loaderUnderTest = ActivatorUtilities.CreateInstance<FileModelLoader>(sp);
+            loaderUnderTest.Start(Path.Combine("TestModels", "SentimentModel.zip"), false);
+
+            var model = loaderUnderTest.GetModel();
```

Comment:
Do you think it's reasonable to make a prediction using the loaded model?

Reply:
Yes. Added.